### PR TITLE
Fix `references(symbol)` for List Binding Patterns

### LINKED ISSUE #33132: Goto definition doesn't work when function call expression returns a tuple
**Description:**
Consider:
```ballerina
function testFunction() {
    string l = "";
    var [fName, fBody] = parseField(l);
}

function parseField(string val) returns [string, int] {
    return [val, 10];
}
```
When goto definition operation is performed on `parseField(l)` inside `testFunction()`, it doesn't work. But references to `parseField()` show that usage.

**Steps to reproduce:**
See description

**Affected Versions:**
SL Beta 4 Snapshot
SL Beta 3

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #33132

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -2814,6 +2814,7 @@ private VariableDefinitionNode createBLangVarDef(Location location,
             case LIST_BINDING_PATTERN:
                 initializeBLangVariable(variable, typedBindingPattern.typeDescriptor(), initializer,
                         qualifierList);
+                variable.pos = location;
```
Comment (mohanvive) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:2817]: Is this fix correct? Don't we need to have a fix like below?
variable.pos = getPosition(bindingPattern);

Comment (mohanvive) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:2817]: Check this again.

**Review by mohanvive — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 1 above)
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:2817]: @mohanvive 

Considering the example `var [fName, fBody] = parseField(l);`, 

- `getPosition(bindingPattern);` would provide the position of `var <start>[fName, fBody]<end> = parseField(l);`

- `getPosition(typedBindingPattern)` would provide the position of `<start>var [fName, fBody]<end> = parseField(l);`

- `location` would provide the position for the entire var declaration node `<start>var [fName, fBody] = parseField(l)<end>`

This approach is also being used for the `WILDCARD_BINDING_PATTERN` case at https://github.com/ballerina-platform/ballerina-lang/blob/8adfc0ac5b12d60c70d33893a15e49e7eba2920e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java#L2798

Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Used a different approach to set the position of `variable` as `var <start>[fName, fBody] = parseField(l);<end>`

**Review by mohanvive — APPROVED** [review]

**Pull request merged**